src/services/webapi/webapi.service.ts

```diff
@@ -18,17 +18,20 @@ export class WizdomWebApiService implements IWizdomWebApiService {
         this.state.corsProxyFailed = false;      
         this.corsProxy.AddHandler("WizdomCorsProxySuccess", (message) => {
             this.state.corsProxyReady = true;
+            this.state.corsProxyFailed = false;
+            this.corsProxy.HandleMessage("WizdomCorsProxyFailed");
             for (var i = 0; i < this.state.deferredQueue.length; i++) {
                 this.makeRequest.apply(this, this.state.deferredQueue[i]);
             }
             this.state.deferredQueue = [];
         });
         this.corsProxy.AddHandler("WizdomCorsProxyFailed", (message) => {
             this.state.corsProxyFailed = true;
-            for (var i = 0; i < this.state.deferredQueue.length; i++) {
-                this.makeRequest.apply(this, this.state.deferredQueue[i]);
-            }
-            this.state.deferredQueue = [];
+            // dont clear the queue on fail, to allow it to recover
+            // for (var i = 0; i < this.state.deferredQueue.length; i++) {
+            //     this.makeRequest.apply(this, this.state.deferredQueue[i]);
+            // }
+            // this.state.deferredQueue = [];
```

Comment:
This means that the corsproxy will never notify anyone that it's failed thus making the whole thing kind of a moot point